pkg/pillar/cmd/zedagent/handlevolume.go

```diff
@@ -128,3 +131,47 @@ func handleVolumeStatusDelete(ctxArg interface{},
 	PublishVolumeToZedCloud(ctx, uuidStr, nil, ctx.iteration)
 	ctx.iteration++
 }
+
+func createVolumeInstanceMetrics(ctx *getconfigContext, reportMetrics *metrics.ZMetricMsg) {
+	log.Debugf("Volume instance metrics started")
+	sub := ctx.subVolumeStatus
+	volumelist := sub.GetAll()
+	if volumelist == nil || len(volumelist) == 0 {
+		return
+	}
+	for _, volume := range volumelist {
+		volumeStatus := volume.(types.VolumeStatus)
+		volumeMetric := new(metrics.ZMetricVolume)
+		volumeMetric.Uuid = volumeStatus.VolumeID.String()
+		volumeMetric.DisplayName = volumeStatus.DisplayName
+		getVolumeResourcesMetrics(volumeStatus, volumeMetric)
+		reportMetrics.Vm = append(reportMetrics.Vm, volumeMetric)
+	}
+	log.Debugf("Volume instance metrics done: %v", reportMetrics.Vm)
+}
+
+func getVolumeResourcesMetrics(volumeStatus types.VolumeStatus,
+	volumeMetric *metrics.ZMetricVolume) error {
+
+	if volumeStatus.IsContainer() {
+		// XXX For container volumes, max size is coming zero
+		// from the controller. So for now, we are setting up
+		// max size and the current size equal to the downloaded size
```

Comment:
Isn't this comment misplaced?

And shouldn't you use the VolumeStatus size info?

Reply:
Comments removed. For metrics and info, we are not copying the sizes from the pubsub.